operator/builtin/output/count/count.go

```diff
@@ -0,0 +1,157 @@
+type CountOutputConfig struct {
+	helper.OutputConfig `yaml:",inline"`
+	Path                string          `json:"path,omitempty" yaml:"path,omitempty"`
+	Duration            helper.Duration `json:"duration,omitempty" yaml:"duration,omitempty"`
+}
+
+var defaultCounterDuration = helper.NewDuration(1 * time.Minute)
+
+func init() {
+	operator.Register("count_output", func() operator.Builder { return NewCounterOutputConfig("") })
+}
+
+// NewCounterOutputConfig creates the default config for the count_output operator.
+func NewCounterOutputConfig(operatorID string) *CountOutputConfig {
+	return &CountOutputConfig{
+		OutputConfig: helper.NewOutputConfig(operatorID, "count_output"),
+		Duration:     defaultCounterDuration,
+	}
+}
+
+// Build will build an instance of the count_output operator
+func (c CountOutputConfig) Build(bc operator.BuildContext) ([]operator.Operator, error) {
+	outputOperator, err := c.OutputConfig.Build(bc)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	counterOperator := &CountOutput{
+		OutputOperator: outputOperator,
+		ctx:            ctx,
+		cancel:         cancel,
+		interval:       c.Duration.Raw(),
+		path:           c.Path,
+		numEntries:     0,
+		wg:             sync.WaitGroup{},
+	}
+
+	return []operator.Operator{counterOperator}, nil
+}
+
+// CountOutput is an output operator
+type CountOutput struct {
+	helper.OutputOperator
+	ctx      context.Context
+	interval time.Duration
+	start    time.Time
+	path     string
+	file     *os.File
+	encoder  *json.Encoder
+	wg       sync.WaitGroup
+	cancel   context.CancelFunc
+
+	numEntries uint64
+}
+
+// Process increments the counter of the output operator
+func (co *CountOutput) Process(_ context.Context, _ *entry.Entry) error {
+	co.numEntries++
+	return nil
+}
+
+// Start begins messaging count output to either stdout or a file
+func (co *CountOutput) Start() error {
+	err := co.determineOutput()
+	if err != nil {
+		return err
+	}
+
+	co.start = time.Now()
+	co.wg.Add(1)
+	go co.startCounting()
+
+	return nil
+}
+
+// Stop tells the CountOutput to stop gracefully
+func (co *CountOutput) Stop() error {
+	co.cancel()
+	co.wg.Wait()
+	if co.file != nil {
+		return co.file.Close()
+	}
+	return nil
+}
+
+func (co *CountOutput) startCounting() {
+	defer co.wg.Done()
+
+	ticker := time.NewTicker(co.interval)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+		case <-co.ctx.Done():
+			return
+		}
+
+		err := co.logCount()
+		if err != nil {
+			return
+		}
+	}
+}
+
+type countObject struct {
+	Entries          uint64  `json:"entries"`
+	ElapsedMinutes   float64 `json:"elapsedMinutes"`
+	EntriesPerMinute float64 `json:"entries/minute"`
+	Timestamp        string  `json:"timestamp"`
+}
+
+func (co *CountOutput) logCount() error {
+	now := time.Now()
+	elapsedMinutes := now.Sub(co.start).Minutes()
+	entriesPerMinute := float64(co.numEntries) / math.Max(elapsedMinutes, 1)
```

Comment:
Is the max check to ensure we don't divide by zero? Seems like if we have  0<`elapsedMinutes` < 1 it would lose that value